### ISSUE TITLE
actix-cors: preflight requests requires Access-Control-Request-Method header

### ISSUE THREAD

**Issue by liangchunn** [actix/actix-extras#225]
## Expected Behavior
Preflight requests (`OPTIONS`) to the server wrapped in the CORS middleware should not require the `Access-Control-Request-Method` header.

## Current Behavior
When executing `curl -X OPTIONS http://0.0.0.0:8080 -i`, the server responds with a 400 and complains that the header `Access-Control-Request-Method` is missing.

AFAIK some browsers don't send the header when requesting a CORS resource, which causes the request to fail.

## Possible Solution
The problem was mentioned in https://github.com/actix/actix-web/issues/702 and fixed in https://github.com/actix/actix-web/pull/816 , but it's not ported over here on actix-extras/actix-cors

## Steps to Reproduce (for bugs)
1. 
```rust
use actix_cors::Cors;
use actix_web::{get, http, web, App, HttpServer, Responder};

#[get("/hello/{name}")]
async fn greet(name: web::Path<String>) -> impl Responder {
    format!("Hello {name}!")
}

#[actix_web::main]
async fn main() -> std::io::Result<()> {
    HttpServer::new(|| {
        let cors = Cors::default()
            .allow_any_origin()
            .allowed_methods(vec!["GET"])
            .allowed_headers(vec![http::header::AUTHORIZATION, http::header::ACCEPT])
            .max_age(3600);
        App::new()
            .wrap(cors)
            .route("/hello", web::get().to(|| async { "Hello World!" }))
            .service(greet)
    })
    .bind(("127.0.0.1", 8080))?
    .run()
    .await
}

```
2. 
```
➜ curl -X OPTIONS http://0.0.0.0:8080 -i
HTTP/1.1 400 Bad Request
content-length: 73
vary: Origin, Access-Control-Request-Method, Access-Control-Request-Headers
date: Mon, 07 Mar 2022 11:00:32 GMT

Request header `Access-Control-Request-Method` is required but is missing%
```

## Your Environment
- rustc 1.59.0 (9d1b2106e 2022-02-23)
- actix-cors 0.6.0
- actix-web 4.0.1

**Comment by liangchunn** [comment]
Thanks for the swift fix! @robjtede

**Comment by robjtede** [comment]
fix released in 0.6.1
